### PR TITLE
Demo examples and managing sub-apps. 

### PR DESCRIPTION
**Pull request by bednarz1990**

### CODE REVIEW

**Comment by bednarz1990** [comment]
This is whole SDK without docs. As you said on daily you would like to see progress so I pushed this branch to watch progress in some way :)

File: Apidaze.SDK.Tests.Integration/XML/Program.cs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                        var buffer = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(e));
+                        response.ContentLength64 = buffer.Length;
+                        response.OutputStream.Write(buffer, 0, buffer.Length);
+
+                        //TODO: Log the exception
+                    }
+                }
+            }
+
+            private static bool GetResponse(HttpListenerContext context, HttpListenerResponse response, bool handled)
```
Comment (shadzik) [inline Apidaze.SDK.Tests.Integration/XML/Program.cs]: The handled parameter is also here not necessary. Also from a logical standpoint, you're evaluating the request http method in a function that has actually nothing to do with the request itself. This method should IMHO, just return the XML and not care about anything related to the http request. The request itself should be evaluated in the `ProcessRequest(HttpListenerContext context)` method. That way, you check for the method just once, and not in every function that returns an XML doc.

**Review by shadzik — CHANGES_REQUESTED** [review]

File: Apidaze.SDK.Tests.Integration/XML/Program.cs (hunk 1 above)
Comment (bednarz1990) [reply Apidaze.SDK.Tests.Integration/XML/Program.cs]: Please see my last commit. It's fixed.

File: Apidaze.SDK.Tests.Integration/IvrExample/WebService.cs [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+{
+    /// <summary>
+    /// Class WebService.
+    /// </summary>
+    internal static class WebService
+    {
+        /// <summary>
+        /// The port the HttpListener should listen on.
+        /// </summary>
+        private static readonly string SERVER_URL = "http://780f6967.ngrok.io";
```
Comment (shadzik) [inline Apidaze.SDK.Tests.Integration/IvrExample/WebService.cs]: Hardcoded ngrok

File: Apidaze.SDK.Tests.Integration/IvrExample/WebService.cs [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+            using var response = context.Response;
+            try
+            {
+                Console.WriteLine(context.Request.RawUrl + "\n" + DateTime.Now);
+                var handled = false;
+                var contextHttp = new ContextHttp(context.Request.Url.AbsolutePath);
+                switch (context.Request.HttpMethod)
+                {
+                    case "GET":
+                        if (contextHttpDictionary.ContainsKey(contextHttp))
```
Comment (shadzik) [inline Apidaze.SDK.Tests.Integration/IvrExample/WebService.cs:192]: That looks good, nicely done!

File: Apidaze.SDK.Tests.Integration/UpdateExternalScriptUrl/Program.cs [hunk 4]
```diff
@@ -25,10 +25,10 @@ static void Main(string[] args)
             var apiFactory = ApplicationManager.CreateApiFactory(new Credentials(apiKey, apiSecret));
 
             // id of script to be fetched
-            var id = 1589L;
+            var id = 1801L;
 
             // new url
-            var newScriptUrl = "http://8a66b0b1.eu.ngrok.io/";
+            var newScriptUrl = "http://c89e3bf5.ngrok.io";
```
Comment (shadzik) [inline Apidaze.SDK.Tests.Integration/UpdateExternalScriptUrl/Program.cs]: Hardcoded ngrok

File: Apidaze.SDK.Tests.Integration/XML/Program.cs [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+            /// <param name="context">The context.</param>
+            /// <param name="response">The response.</param>
+            /// <returns><c>true</c> if handled, <c>false</c> otherwise.</returns>
+            private static bool GetResponse(HttpListenerContext context, HttpListenerResponse response)
+            {
+                var handled = false;
+                switch (context.Request.HttpMethod)
+                {
+                    case "GET":
+                        var script = ApidazeScript.Build();
```
Comment (shadzik) [inline Apidaze.SDK.Tests.Integration/XML/Program.cs]: This  `context.Request.HttpMethod` should also be moved into `ProcessRequest`

**Review by shadzik — CHANGES_REQUESTED** [review]

File: Apidaze.SDK.Tests.Integration/IvrExample/WebService.cs (hunk 2 above)
Comment (bednarz1990) [reply Apidaze.SDK.Tests.Integration/IvrExample/WebService.cs]: ok.

File: Apidaze.SDK.Tests.Integration/IvrExample/WebService.cs (hunk 3 above)
Comment (bednarz1990) [reply Apidaze.SDK.Tests.Integration/IvrExample/WebService.cs:192]: Thx. Happy to help.

File: Apidaze.SDK.Tests.Integration/UpdateExternalScriptUrl/Program.cs (hunk 4 above)
Comment (bednarz1990) [reply Apidaze.SDK.Tests.Integration/UpdateExternalScriptUrl/Program.cs]: ok

File: Apidaze.SDK.Tests.Integration/XML/Program.cs (hunk 5 above)
Comment (bednarz1990) [reply Apidaze.SDK.Tests.Integration/XML/Program.cs]: ok

**Review by shadzik — APPROVED** [review]

**Review by barinali — APPROVED** [review]

**Pull request merged**